frontend/src/components/data-table/column-header.tsx

```diff
@@ -343,23 +372,180 @@ const TextFilter = <TData, TValue>({
+      <>
+        <Table className="border-collapse text-sm overflow-auto block max-h-64">
+          <TableHeader>
+            <TableRow className="hover:bg-transparent py-0">
+              <TableHead className="w-7 py-0 px-2">
+                <Checkbox
+                  checked={chosenValues.length === filteredData.length}
+                  onCheckedChange={(checked) => {
+                    if (typeof checked === "string") {
+                      return;
+                    }
+                    toggleAllCheckbox(checked);
+                  }}
+                  aria-label="Select all"
+                />
+              </TableHead>
+              <TableHead className="text-foreground font-bold py-0">
+                {column.id}
+              </TableHead>
+              <TableHead className="text-foreground font-bold py-0">
+                count
+              </TableHead>
+            </TableRow>
+          </TableHeader>
+          <TableBody>
+            {filteredData.map((row, rowIndex) => {
+              const value = row[0];
+              const count = row[1];
+              return (
+                <TableRow key={rowIndex}>
+                  <TableCell>
+                    <Checkbox
+                      checked={chosenValues.includes(value)}
+                      onCheckedChange={(checked) => {
+                        if (typeof checked === "string") {
+                          return;
+                        }
+                        handleCheckboxClick(checked, value);
+                      }}
+                      aria-label="Select row"
+                    />
+                  </TableCell>
+                  <TableCell className="overflow-hidden w-48 max-w-48 max-h-20 line-clamp-3">
+                    {String(value)}
+                  </TableCell>
+                  <TableCell>{count}</TableCell>
+                </TableRow>
+              );
+            })}
+          </TableBody>
+        </Table>
+        <FilterButtons
+          onApply={handleApply}
+          onClear={() => {
+            setChosenValues([]);
+          }}
+          clearButtonDisabled={chosenValues.length === 0}
+        />
+      </>
+    );
+  }
+
+  return (
+    <Popover open={true}>
+      <PopoverTrigger />
+      <PopoverContent className="w-80">
+        <PopoverClose className="absolute top-2 right-2">
+          <Button
+            variant="link"
+            size="sm"
+            onClick={() => setIsSetFilterOpen(false)}
+          >
+            X
+          </Button>
+        </PopoverClose>
+        <div className="flex flex-col gap-1.5 pt-1">
+          <span className="text-sm font-semibold mx-auto mb-2">Set Filter</span>
+          <Input
+            value={query}
+            onChange={(e) => setQuery(e.target.value)}
+            placeholder="Search"
+            className="my-0 py-1"
+            autoFocus={true}
+          />
+          {dataTable}
+        </div>
+      </PopoverContent>
```

Comment:
Small UX suggestion: if we can use a `<Combobox`, we can get keyboard navigation to work correctly.

it would mean you couldn't use the table, but you can still format each `ComboItem` the same way.